internal/infrastructure/middleware/validate_api.go

```diff
@@ -285,19 +289,15 @@ func RequestResponseValidatorWithConfig(config *RequestResponseValidatorConfig)
 						responseValidationInput,
 					); err != nil {
 						// write error response
-						c.Response().Header().Set(echo.HeaderContentType, "text/plain")
-						c.String(http.StatusInternalServerError, err.Error())
-					} else {
-						// Write original response
-						rw.WriteHeader(resRec.status)
-						resRec.buffer.WriteTo(rw)
+						return echo.NewHTTPError(http.StatusBadRequest, err.Error())
```

Comment:
I think that this should stay as `StatusInternalServerError` . This is a *reseponse* validation error; the request would already have been validated.

Reply:
Fixed. Thanks.